x-pack/plugins/observability_solution/observability_ai_assistant_app/server/functions/query/correct_common_esql_mistakes.test.ts

```diff
@@ -11,147 +11,160 @@ describe('correctCommonEsqlMistakes', () => {
+  function expectQuery({ input, expectedOutput }: { input: string; expectedOutput: string }) {
     expect(normalize(correctCommonEsqlMistakes(input).output)).toEqual(normalize(expectedOutput));
   }
 
   it('replaces aliasing via the AS keyword with the = operator', () => {
-    expectQuery(`FROM logs-* | STATS COUNT() AS count`, 'FROM logs-*\n| STATS count = COUNT()');
-
-    expectQuery(`FROM logs-* | STATS COUNT() as count`, 'FROM logs-*\n| STATS count = COUNT()');
-
-    expectQuery(
-      `FROM logs-* | STATS AVG(transaction.duration.histogram) AS avg_request_latency, PERCENTILE(transaction.duration.histogram, 95) AS p95`,
-      `FROM logs-*
-      | STATS avg_request_latency = AVG(transaction.duration.histogram), p95 = PERCENTILE(transaction.duration.histogram, 95)`
-    );
-
-    expectQuery(
-      `FROM traces-apm*
+    expectQuery({
+      input: `FROM logs-* | STATS COUNT() AS count`,
+      expectedOutput: 'FROM logs-*\n| STATS count = COUNT()',
+    });
+
+    expectQuery({
+      input: `FROM logs-* | STATS COUNT() as count`,
+      expectedOutput: 'FROM logs-*\n| STATS count = COUNT()',
+    });
+
+    expectQuery({
+      input: `FROM logs-* | STATS AVG(transaction.duration.histogram) AS avg_request_latency, PERCENTILE(transaction.duration.histogram, 95) AS p95`,
+      expectedOutput: `FROM logs-*
+      | STATS avg_request_latency = AVG(transaction.duration.histogram), p95 = PERCENTILE(transaction.duration.histogram, 95)`,
+    });
+
+    expectQuery({
+      input: `FROM traces-apm*
       | WHERE @timestamp >= NOW() - 24 hours
       | STATS AVG(transaction.duration.us) AS avg_duration, SUM(success) AS total_successes, COUNT(*) AS total_requests BY service.name`,
-      `FROM traces-apm*
+      expectedOutput: `FROM traces-apm*
       | WHERE @timestamp >= NOW() - 24 hours
-      | STATS avg_duration = AVG(transaction.duration.us), total_successes = SUM(success), total_requests = COUNT(*) BY service.name`
-    );
+      | STATS avg_duration = AVG(transaction.duration.us), total_successes = SUM(success), total_requests = COUNT(*) BY service.name`,
+    });
   });
-
-  it(`replaces " or ' escaping in FROM statements with backticks`, () => {
-    expectQuery(`FROM "logs-*" | LIMIT 10`, 'FROM logs-*\n| LIMIT 10');
-    expectQuery(`FROM 'logs-*' | LIMIT 10`, 'FROM logs-*\n| LIMIT 10');
-    expectQuery(`FROM logs-* | LIMIT 10`, 'FROM logs-*\n| LIMIT 10');
+  it("replaces ` or ' escaping in FROM statements with double quotes", () => {
+    expectQuery({ input: `FROM "logs-*" | LIMIT 10`, expectedOutput: 'FROM "logs-*"\n| LIMIT 10' });
+    expectQuery({ input: `FROM 'logs-*' | LIMIT 10`, expectedOutput: 'FROM "logs-*"\n| LIMIT 10' });
+    expectQuery({
+      input: `FROM 'logs-2024-07-01','logs-2024-07-02' | LIMIT 10`,
+      expectedOutput: 'FROM "logs-2024-07-01","logs-2024-07-02"\n| LIMIT 10',
+    });
+    expectQuery({ input: `FROM logs-* | LIMIT 10`, expectedOutput: 'FROM logs-*\n| LIMIT 10' });
```

Comment:
can we add one here for backticks as well? i see the LLM do this all the time now 😄